docs/references/public-en.md

```diff
@@ -1,28 +1,88 @@
+# Public JSON RPC Endpoints
 
-By using node services, you can focus on developing your application or product on Kaia instead of spending your resources on node operation. Below is a list of node providers.
+Publicly exposed JSON-RPC endpoints allow you to test and run your blockchain products by providing interaction with the Kaia network without running your own node.
 
-## Kaia API Service (KAS)
+Running your own Kaia Endpoint Node (EN) is not simple, it requires technical expertise, monitoring and computing resources. It comes with a cost of maintaining storage, network bandwidth as well as having to divert engineering time and resources; nodes must be kept up to date and health checked regularly.
+
+Hence, the main benefit of using an existing Public EN is that it allows you to solely focus on building and testing your blockchain product without the distraction of maintaining infrastructure to connect and interact with the Kaia network.
+ 
+## Things to Consider
+
+- The node providers are not responsible for any damage or losses caused in relation to traffic or interaction with the nodes.
+- If traffic is concentrated on certain nodes, you may experience service delay.
+- To prevent too many requests, rate limits may apply on a per-node basis, which are subject to change without prior notification.
+
+## Public JSON-RPC Endpoints
+
+Below is the list of the network domains offered by Kaia’s Public Node Providers.
+
+### Mainnet Public JSON-RPC Endpoints
+
+Please keep in mind that these endpoints are provided to the community for testing and development purposes.
+Since we cannot guarantee uptime and stability of the endpoints, do not use them for commercial purposes.
+
+**HTTPS**
+
+|Service Provider|Endpoints|Namespaces|Type|
+|---|---|---|---|
+|[Kaia Foundation](https://www.klaytn.foundation)|`https://public-en-cypress.klaytn.net`|klay,eth,net|Full|
+||`https://archive-en.cypress.klaytn.net`|klay,eth,net|Archive|
+|[All That Node](www.allthatnode.com)|`https://klaytn-mainnet.g.allthatnode.com/full/evm`|klay,eth,net|Full|
+|[BlockPI Network](https://blockpi.io/)|`https://klaytn.blockpi.network/v1/rpc/public`|klay,eth,net|Full|
+|[OnFinality](https://onfinality.io/)|`https://klaytn.api.onfinality.io/public`|klay,eth,net|Full|
+|[Pokt Network](https://pokt.network/)|`https://klaytn-rpc.gateway.pokt.network/`|klay,eth,net|Full|
+
+**WebSocket** 
+
+|Service Provider|Endpoints|Namespaces|Type|
+|---|---|---|---|
+|[Kaia Foundation](https://www.klaytn.foundation)|`wss://public-en-cypress.klaytn.net/ws`|klay,eth,net|Full|
+||`wss://archive-en.cypress.klaytn.net/ws`|klay,eth,net|Archive|
+|[OnFinality](https://onfinality.io/)|`wss://klaytn.api.onfinality.io/public-ws`|klay,eth,net|Full|
+
+### Testnet (Kairos) Public JSON-RPC Endpoints
+
+**HTTPS**
+
+|Service Provider|Endpoints|Namespaces|Type|
+|---|---|---|---|
+|[Kaia Foundation](https://www.klaytn.foundation)|`https://public-en-baobab.klaytn.net`|klay,eth,net|Full|
```

Comment:
Hi @scott-klaytn , can we also please consider adding kairos endpoint here along with others ?
https://public-en.kairos.node.kaia.io
Reference: https://klaytn.foundation/kaia-testnet-kairos-launch/

Reply:
Hi @scott-klaytn, can we leave TxError page as is? or better still add to the common sidebar?

imo, It would fit the new position only if it list the possible errors emitted when using caver.js or any of the libraries. In this case, its just the general tx error code when deploying, interacting with a contract or using any of the libraries on Kaia. 

Do lemme know your thoughts. Thanks.

Reply:
I've replaced the old testnet endpoint with the new one. For others, I will update them as new urls are made available practically. @praveen-klaytn